chia/wallet/wallet_node.py

```diff
@@ -201,9 +219,9 @@ async def _start(
         self._new_peak_queue = NewPeakQueue(inner_queue=asyncio.PriorityQueue())
 
         self.synced_peers = set()
-        private_key = await self.get_key_for_fingerprint(fingerprint)
+        private_key = await self.get_private_key(fingerprint or self.get_last_used_fingerprint(), True)
```

Comment:
```suggestion
        if fingerprint is None:
            fingerprint = self.get_last_used_fingerprint()
        private_key = await self.get_private_key(fingerprint)
        # If the key for the fingerprint we provided doesn't exist just try to get the first key
        if private_key is None and fingerprint is not None:
            private_key = await self.get_private_key(None)
```

You can just drop the `use_first_key_if_not_found` parameter and all the related code if you use the above unless i miss something.

Reply:
I could do this, but it moves the complexity to the caller instead of making it an implementation detail of `get_private_key()`. The behavior is the same, yes, but I don't see a reason to make `_start()` more complicated than it already is. Seems like just a stylistic difference of opinion.

Reply:
Hm, so right now you are adding complexity to `get_private_key` by adding the parameter `use_first_key_if_not_found` which kind of re-implements an already existing behaviour of `get_private_key`
inside the function itself only because its needed in one place. Maybe it's just me but i rather would avoid this. 

Would you now also add another parameter `alternative_fingerprint` to try fingerprint `B` in case fetching `A` fails if you need this in some place? This would blow the function up more and more. 

But ok, if you really rather want to stick with `use_first_key_if_not_found` then maybe at least drop `retry_with_first_key` and just do the fun inside the `except` block based on `use_first_key_if_not_found` or was there a reason for moving it down?